alembic/versions/2023_04_25_1342-c5920d3da32f_creation_of_the_events_and_badges_tables.py

```diff
@@ -0,0 +1,65 @@
+"""Creation of the events and badges tables
+
+Revision ID: c5920d3da32f
+Revises: 
+Create Date: 2023-04-25 13:42:45.102279
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "c5920d3da32f"
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "badges",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("user_id", sa.String(), nullable=True),
+        sa.Column("device_id", sa.String(), nullable=True),
+        sa.Column("badge_name", sa.String(), nullable=True),
+        sa.Column("level", sa.Integer(), nullable=True),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(op.f("ix_badges_device_id"), "badges", ["device_id"], unique=False)
+    op.create_index(op.f("ix_badges_user_id"), "badges", ["user_id"], unique=False)
+    op.create_table(
+        "events",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("user_id", sa.String(), nullable=True),
+        sa.Column("device_id", sa.String(), nullable=True),
+        sa.Column("event_type", sa.String(), nullable=True),
+        sa.Column(
+            "timestamp",
+            sa.DateTime(),
+            server_default=sa.text("(CURRENT_TIMESTAMP)"),
+            nullable=True,
+        ),
+        sa.Column("barcode", sa.String(), nullable=True),
```

Comment:
in our first migration (that we will fake in production), we should have barcode still as an Integer.

Then you should generate a second migration to have barcode turned to String.

Reply:
The migration should also take care of migrating codes:

if you have 8 chars or less  --> pad in front with 0 up to 8 chars
otherwise --> pad in front with 0 up to 13 chars